app/controllers/users_controller.rb

```diff
@@ -36,6 +37,12 @@ def update
 
   private
 
+  def authorize_password_login
+    return if Feature.authenticate_with_password?
+
+    halt 403
```

Comment:
This is the most simple way to show a 403

Reply:
Alternate: Like `ensure_registration`: Redirect to `redirect_to root_path`

Reply:
How about we rename the filter and check them both in one:

`before_action :ensure_registration_allowed`

then:

```
def ensure_registration_allowed
  if Feature.disabled?(:registration) || Feature.disabled?(:authenticate_with_password)
    redirect_to ...
  end
end
```

(I added disabled? in my other PR)